Crypto/Random/Types.hs

```diff
@@ -5,16 +5,21 @@
 -- Stability   : experimental
 -- Portability : Good
 --
+{-# LANGUAGE ScopedTypeVariables #-}
 module Crypto.Random.Types
     (
       MonadRandom(..)
     , MonadPseudoRandom
     , DRG(..)
+    , PRG(..)
+    , prgNewEntropy
+    , prgNew
```

Comment:
I would prefer not to expose prgNewEntropy unless absolutely necessary. This function relies on its argument always returning requested length.

Reply:
I can remove it, but it's really just a convenience wrapper around `prgNewSeed`, a typeclass methods of the new `PRG` class I'm adding in this PR. Anyone can recreate `prgNewEntropy` themselves, so I'm not sure hiding it achieves anything in terms of avoiding misuse.

For `prgNewSeed`, I basically copied the type signature of the existing `ChaCha.initializeSimple`. Its implementation checks the length of the seed, but the type signature does not make this obvious. So I could modify both of these to return a `CryptoFailable StateSimple/gen` instead - and then this type would propagate through into `prgNewEntropy`. How does that sound? That should make it much more obvious that anyone implementing `prgNewSeed` should explicitly check the length.

Reply:
For example, in 992207539907d343b2d6435da1407e96b6353fda

Reply:
I still think the real convenience wrapper is `prgNew`, without a `CryptoFailable` result.
`MonadRandom` is supposed to always return the requested length.
So if the `PRG` and `MonadRandom` instances are implemented correctly, a result _will_ be generated.

Misuse comes only with `prgNewEntropy`.
People not wanting `getRandomBytes`, or wanting length larger than minimum, can call the `PRG` methods directly.
They will know best if the length requirement is always met or not.

Reply:
In the interests of getting this PR merged, I don't want to continue arguing over a minor point, and so I've un-exposed `prgNewEntropy` as you wanted, and made `prgNew` auto-unwrap the `CryptoFailable`.

However, I still believe the old version is OK and more helpful for users. In Haskell, typeclass instances are chosen by the caller, so I don't see a fundamental usage/abuse/security difference between:

1. letting the user implicitly select the instance of `MonadRandom` class in `prgNew`, vs
2. letting the user explicitly select the `getEntropy` function in `prgNewEntropy`

Either of these could return the wrong length, but (a) the programmer can easily check the function they're using behaves correctly and (b) since 9922075 we mandate a length check in `prgNewSeed`.

Arguably, the explicit version (2) is easier to check - with `MonadRandom` (1) you have to look up the instance definition first, which could be a redirection to another function (e.g. `MonadRandom IO`), so it takes two steps. This is why in my own code, I prefer to use `getEntropy` explicitly and avoid the implicit `MonadRandom IO` instance.

Reply:
I don't want to merge `initializeSimpleErr` like this.
Bytes after the 40th are ignored and don't contribute to the DRG.

Reply:
@ocheron It is the same as the current behaviour of `initializeSimple` - as you can see in the diff I did not change the behaviour. How do you want to fix it, and shouldn't it be done in a separate PR?